### DISCUSSION TITLE
Layout tab & design idea

### DISCUSSION THREAD

**Discussion by aetkie** [UNC-Project-Open-AAC/OS-DPI#156] [Ideas]
Here's a wish:

When designing, I am often unsure which component of the Layout is showing up in the design. I wish that when I hovered over a component in the design, it's counterpart in the Layout was highlighted. For instance, if I hover over a button, that button is highlighted in the Layout tab. Or vice versa. Would that be helpful for anyone else?

**Comment by gbishop** [comment]
Excellent suggestion. The previous release had that feature and  it got lost in the upgrade. I should make it work again.

**Reply by aetkie** [reply]
That would be great! Thank you!

**Reply by jbizovi** [reply]
Yes @aetkie, I just noticed the same thing! This is necessary for my brain.

**Comment by gbishop** [comment]
What should happen when you are working on a child of a tab panel that is
not currently visible in the UI?

**Reply by jbizovi** [reply]
Let me think about that one...

**Comment by gbishop** [comment]
It seems to me the possibilities are:

1. Climb up the tree to the first visible ancestor and highlight it. So the
parent tab control would be highlighted in the example I sighted.
2. Force the panel to be visible. This is tricky because it requires
updating the state which could lead to surprising results because now the
behavior of the UI depends in a non-obvious way on where you are in the
editor. This was a source of confusion with the previous implementation.
3.Somehow make the panel visible for a few seconds. Could get really ugly
when moving quickly through the layout.
4. Maybe add a command to one of the menus to make the current panel
visible. I often keep the tab buttons visible during development for just
this reason. A command alternative could be useful.

The same goes for modal dialogs that are currently invisible.

**Reply by jbizovi** [reply]
I feel like it used to force the panel to be visible? Maybe I'm just imagining that. I get lost in even the simplest designs though so having this back would be great.

**Comment by gbishop** [comment]
It did force the panel to be visible and resulted in confusion because the
UI behaved differently depending on your current focus in the editor. Maybe
it only happens on the transition into?

**Comment by gbishop** [comment]
The production code now has this feature enabled. See #157

**Reply by gbishop** [reply]
I don't like the Control-h binding for this... What should it be?
